src/ceph-volume/ceph_volume/util/system.py

```diff
@@ -275,7 +275,22 @@ def get_mounts(devices=False, paths=False, realpath=False):
         return paths_mounted
 
 
-def set_context(path, recursive = False):
+def set_context(path, recursive=False):
+    """
+    Calls ``restorecon`` to set the proper context on SELinux systems. Only if
+    the ``restorecon`` executable is found anywhere in the path it will get
+    called.
+
+    If the ``CEPH_VOLUME_SKIP_RESTORECON`` environment variable is set to any
+    of: "1", "true", "yes" the call will be skipped as well.
+    """
+
+    skip = os.environ.get('CEPH_VOLUME_SKIP_RESTORECON', '')
```

Comment:
You could also look at the `selinuxenabled` binary -- whether it exits with failure or doesn't exist at all -- you can just skip calling restorecon in that case.

In the ceph container, the binary exists but exits with 1 suggesting that SELinux is not enabled (even though it is in the host system).

Reply:
@b-ranto do you think this should go in as an addition? or instead of the environment variable?

Reply:
@alfredodeza I don't think we would need it any more although it wouldn't hurt to have the variable anyway. I would decide on whether to have/not have it based on the consistency -- do we use/have more env variables like these in ceph-volume code? If we do I'd keep it. If we don't I'd use just the `selinuxenabled` method.

Reply:
we have a couple, no more than 4 I think.  But your suggestion makes sense, it might even cause rook not to have to pass an environment variable. Can you take another look?

Reply:
We will still need the env var for Rook because in our case Selinux was enabled so for some env it sill makes sense to have the ability to skip selinux commands.

Reply:
@leseb yes, I am keeping the environment variable. Could you also confirm if `selinuxenabled` was exiting with a `0` ?

Reply:
```
$ oc exec rook-ceph-operator-7775886dd4-pxls5 -ti bash
bash-4.4$ selinuxenabled
bash-4.4$ echo $?
1
```

Reply:
@leseb so in this case the environment variable wouldn't be needed. We are checking for a non-zero exit status to *skip* the call. I am fine keeping the environment variable though...

Reply:
Yes, because the OSD container is getting a lot more privileged these days and it's not impossible that the command could return 0...
I don't have an OSD on OCS at the moment so I can't test... But keeping the env var sure doesn't hurt.

Also, I think the env var should take precedence over the `selinuxenabled` check.

Reply:
I think either way is fine, however if you go the exit code route, it'd be great if you would add a check if the binary exists at all for non-selinux systems.

I'm aware that `process.run` handles that case, but the resulting log message can be confusing, cp. https://tracker.ceph.com/issues/41802

Reply:
@jan--f the log message is going to end up in the log file, I am omitting returning anything to the terminal for the user.

@leseb I will move the env var to the top so that it takes precedence. I agree that if the caller is explicitly asking not to run anything SELinux then it shouldn't attempt to further verify. Thanks for the feedback.

Reply:
> @jan--f the log message is going to end up in the log file, I am omitting returning anything to the terminal for the user.

I know but it still confuses user that the log complains about SELinux specific binaries on a non-SESLinux system.